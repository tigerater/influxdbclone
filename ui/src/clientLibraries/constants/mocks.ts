export interface ClientLibrary {
  id: string
  name: string
  description?: string
}

export const mockClientLibraries: ClientLibrary[] = [
  {
    id: 'client-lib-js',
    name: 'Javascript',
    description: 'Write to InfuxDB using Javascript',
  },
  {
    id: 'client-lib-node',
    name: 'Node',
    description: 'Write to InfuxDB using Node',
  },
  {
    id: 'client-lib-go',
    name: 'GO',
    description: 'Write to InfuxDB using GO',
  },
  {
    id: 'client-lib-python',
    name: 'Python',
    description: 'Write to InfuxDB using Python',
  },
]
