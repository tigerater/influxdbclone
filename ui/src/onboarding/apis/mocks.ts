import {
  getTelegrafConfigsResponse,
  createTelegrafConfigResponse,
  setSetupParamsResponse,
} from 'mocks/dummyData'

const telegrafsGet = jest.fn(() =>
  Promise.resolve(getTelegrafConfigsResponse.data)
)
const telegrafsPost = jest.fn(() =>
  Promise.resolve(createTelegrafConfigResponse.data)
)
const telegrafsTelegrafIDPut = jest.fn(() =>
  Promise.resolve(createTelegrafConfigResponse.data)
)
const setupPost = jest.fn(() => Promise.resolve(setSetupParamsResponse))
const setupGet = jest.fn(() => Promise.resolve({data: {allowed: true}}))

export const telegrafsAPI = {
  telegrafsGet,
  telegrafsPost,
  telegrafsTelegrafIDPut,
}

export const client = {
  telegrafConfigs: {
    getAll: telegrafsGet,
    getAllByOrg: telegrafsGet,
    create: telegrafsPost,
  },
}

export const setupAPI = {
  setupPost,
  setupGet,
}
