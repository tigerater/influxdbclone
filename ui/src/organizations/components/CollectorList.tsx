// Libraries
import React, {PureComponent} from 'react'
import _ from 'lodash'

// Components
import {IndexList} from 'src/clockface'
import CollectorRow from 'src/organizations/components/CollectorRow'

// DummyData
import {Telegraf} from '@influxdata/influx'
import {getDeep} from 'src/utils/wrappers'

interface Props {
  collectors: Telegraf[]
  emptyState: JSX.Element
  onDelete: (telegrafID: string) => void
  onUpdate: (telegraf: Telegraf) => void
  onOpenInstructions: (telegrafID: string) => void
  onOpenTelegrafConfig: (telegrafID: string, telegrafName: string) => void
}

export default class CollectorList extends PureComponent<Props> {
  public render() {
    const {emptyState} = this.props

    return (
      <>
        <IndexList>
          <IndexList.Header>
            <IndexList.HeaderCell columnName="Name" width="50%" />
            <IndexList.HeaderCell columnName="Bucket" width="50%" />
          </IndexList.Header>
          <IndexList.Body columnCount={3} emptyState={emptyState}>
            {this.collectorsList}
          </IndexList.Body>
        </IndexList>
      </>
    )
  }

  public get collectorsList(): JSX.Element[] {
    const {
      collectors,
      onDelete,
      onUpdate,
      onOpenInstructions,
      onOpenTelegrafConfig,
    } = this.props

    if (collectors !== undefined) {
      return collectors.map(collector => (
        <CollectorRow
          key={collector.id}
          collector={collector}
          bucket={getDeep<string>(collector, 'plugins.0.config.bucket', '')}
          onDelete={onDelete}
          onUpdate={onUpdate}
          onOpenInstructions={onOpenInstructions}
          onOpenTelegrafConfig={onOpenTelegrafConfig}
        />
      ))
    }
    return
  }
}
