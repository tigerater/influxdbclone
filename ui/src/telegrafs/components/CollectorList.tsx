// Libraries
import React, {PureComponent} from 'react'
import _ from 'lodash'
import memoizeOne from 'memoize-one'

// Components
import {ResourceList} from '@influxdata/clockface'
import CollectorRow from 'src/telegrafs/components/CollectorCard'

// Types
import {ITelegraf as Telegraf} from '@influxdata/influx'
import {Sort} from '@influxdata/clockface'
import {SortTypes, getSortedResources} from 'src/shared/utils/sort'

//Utils
import {getDeep} from 'src/utils/wrappers'

type SortKey = keyof Telegraf | 'bucket'

interface Props {
  collectors: Telegraf[]
  emptyState: JSX.Element
  onDelete: (telegraf: Telegraf) => void
  onUpdate: (telegraf: Telegraf) => void
  onFilterChange: (searchTerm: string) => void
  sortKey: string
  sortDirection: Sort
  sortType: SortTypes
  onClickColumn: (nextSort: Sort, sortKey: SortKey) => void
}

export default class CollectorList extends PureComponent<Props> {
  private memGetSortedResources = memoizeOne<typeof getSortedResources>(
    getSortedResources
  )

  public render() {
    const {emptyState, sortKey, sortDirection, onClickColumn} = this.props

    return (
      <>
        <ResourceList>
          <ResourceList.Header>
            <ResourceList.Sorter
              sortKey={this.headerKeys[0]}
              sort={sortKey === this.headerKeys[0] ? sortDirection : Sort.None}
              name="Name"
              onClick={onClickColumn}
            />
            <ResourceList.Sorter
              name="Bucket"
              sortKey={this.headerKeys[1]}
              sort={sortKey === this.headerKeys[1] ? sortDirection : Sort.None}
              onClick={onClickColumn}
            />
          </ResourceList.Header>
          <ResourceList.Body emptyState={emptyState}>
            {this.collectorsList}
          </ResourceList.Body>
        </ResourceList>
      </>
    )
  }

  private get headerKeys(): SortKey[] {
    return ['name', 'bucket']
  }

  public get collectorsList(): JSX.Element[] {
    const {
      collectors,
      sortKey,
      sortDirection,
      sortType,
      onDelete,
      onUpdate,
      onFilterChange,
    } = this.props
    const sortedCollectors = this.memGetSortedResources(
      collectors,
      sortKey,
      sortDirection,
      sortType
    )

    if (collectors !== undefined) {
      return sortedCollectors.map(collector => (
        <CollectorRow
          key={collector.id}
          collector={collector}
          bucket={getDeep<string>(collector, 'plugins.0.config.bucket', '')}
          onDelete={onDelete}
          onUpdate={onUpdate}
          onFilterChange={onFilterChange}
        />
      ))
    }
  }
}
