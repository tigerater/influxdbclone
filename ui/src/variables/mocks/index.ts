// Types
import {Variable} from 'src/types'

export const createVariable = (
  name: string,
  query: string,
  selected?: string
): Variable => ({
  name,
  id: name,
  orgID: 'howdy',
  selected: selected ? [selected] : [],
  labels: [],
  arguments: {
    type: 'query',
    values: {
      query,
      language: 'flux',
    },
  },
})

export const createMapVariable = (
  name: string,
  map: {[key: string]: string} = {},
  selected?: string
): Variable => ({
  name,
  id: name,
  orgID: 'howdy',
  selected: selected ? [selected] : [],
  labels: [],
  arguments: {
    type: 'map',
    values: {...map},
  },
})
