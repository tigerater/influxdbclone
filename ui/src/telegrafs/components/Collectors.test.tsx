// Libraries
import React from 'react'
import {shallow} from 'enzyme'

// Components
import CollectorList from 'src/telegrafs/components/CollectorList'

// Constants
import {telegraf} from 'mocks/dummyData'

const setup = (override?) => {
  const props = {
    collectors: telegraf,
    emptyState: <></>,
    onDelete: jest.fn(),
    onUpdate: jest.fn(),
    onOpenInstructions: jest.fn(),
    onOpenTelegrafConfig: jest.fn(),
    ...override,
  }

  const wrapper = shallow(<CollectorList {...props} />)

  return {wrapper}
}

describe('CollectorList', () => {
  describe('rendering', () => {
    it('renders', () => {
      const {wrapper} = setup()
      expect(wrapper.exists()).toBe(true)
      expect(wrapper).toMatchSnapshot()
    })
  })
})
