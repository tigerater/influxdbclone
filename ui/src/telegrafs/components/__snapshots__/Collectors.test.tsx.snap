// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectorList rendering renders 1`] = `
<Fragment>
  <ResourceList
    testID="resource-list"
  >
    <ResourceListHeader
      testID="resource-list--header"
    >
      <ResourceListSorter
        name="Name"
        sort="none"
        sortKey="name"
        testID="resource-list--sorter"
      />
      <ResourceListSorter
        name="Bucket"
        sort="none"
        sortKey="bucket"
        testID="resource-list--sorter"
      />
    </ResourceListHeader>
    <ResourceListBody
      emptyState={<React.Fragment />}
      testID="resource-list--body"
    >
      <Connect(withRouter(CollectorRow))
        bucket=""
        collector={
          Object {
            "id": "03636a150fb51000",
            "name": "Name this Configuration",
            "orgID": "03636a0aabb51000",
          }
        }
        key="03636a150fb51000"
        onDelete={[MockFunction]}
        onUpdate={[MockFunction]}
      />
      <Connect(withRouter(CollectorRow))
        bucket=""
        collector={
          Object {
            "id": "03636a150fb51001",
            "name": "Name this Configuration",
            "orgID": "03636a0aabb51000",
          }
        }
        key="03636a150fb51001"
        onDelete={[MockFunction]}
        onUpdate={[MockFunction]}
      />
    </ResourceListBody>
  </ResourceList>
</Fragment>
`;
